src/components/CreateCriteriaForm/CreateCreteriaForm.tsx

```diff
@@ -0,0 +1,272 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import styled from 'styled-components';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { ComboBox, Button, AddIcon, Form, GoalIcon, nullable, FormInput } from '@taskany/bricks';
+import { gray7, gray8 } from '@taskany/colors';
+import { SubmitHandler, UseFormRegister, UseFormRegisterReturn, UseFormSetError, useForm } from 'react-hook-form';
+
+import { AddCriteriaScheme, criteriaSchema } from '../../schema/criteria';
+import { Table, TableRow, TitleItem, ContentItem, Title, TextItem, TitleContainer } from '../Table';
+import { StateDot } from '../StateDot';
+import { errorsProvider } from '../../utils/forms';
+
+import { tr } from './CreateCriteriaForm.i18n';
+
+const maxPossibleWeigth = 100;
+
+const StyledPlainButton = styled(Button)`
+    background-color: unset;
+    border: none;
+    padding: 0;
+
+    color: ${gray8};
+
+    border-radius: 0;
+
+    &:hover:not([disabled]),
+    &:focus:not([disabled]) {
+        background-color: unset;
+    }
+
+    &:active:not([disabled]) {
+        transform: none;
+    }
+`;
+
+const StyledTableResults = styled(Table)`
+    grid-template-columns: 30px 210px 40px minmax(max-content, 120px);
+    width: fit-content;
+`;
+
+const StyledFormInput = styled(FormInput)`
+    font-size: 14px;
+    font-weight: normal;
+    padding: 5px 10px;
+
+    border: 1px solid ${gray7};
+    box-sizing: border-box;
+
+    & ~ div {
+        top: 50%;
+    }
+`;
+
+const StyledTableRow = styled.div`
+    display: grid;
+    padding-left: 14px;
+    grid-template-columns:
+        minmax(calc(300px + 10px /* gap of sibling table */), 30%)
+        repeat(2, max-content);
+`;
+
+const StyledSubmitButton = styled(Button)`
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+`;
+
+const GoalSuggestItem = (props: any): React.ReactElement => {
```

Comment:
Because that component has `NextLink` wrapper and by the click user will be navigate to goal page, also i think component have to much more data about goal for this case